src/Core/Request/AbstractApiRequest.php

```diff
@@ -255,9 +255,9 @@ public function map(array $data, Context $context = null, MapperInterface $mappe
      */
     public function executeWithClient(Client $client, array $headers = null)
     {
-        if (!is_null($headers)) {
-            $this->headers = array_merge($headers, $this->headers);
+        if (!is_null($this->externalUserId)) {
+            $headers[self::EXTERNAL_USER_HEADER] = $this->externalUserId;
```

Comment:
method signature should be changed here to default to an empty array. Also before setting the header please check that it's not already set in the given headers array.